query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -59,15 +81,29 @@ class CountOverTimeFunction extends RangeFunction {
   override def apply(startTimestamp: Long, endTimestamp: Long, window: Window,
                      sampleToEmit: TransientRow,
                      queryConfig: QueryConfig): Unit = {
-    sampleToEmit.setValues(endTimestamp, count)
+    sampleToEmit.setValues(endTimestamp, count.toDouble)
   }
 }
 
-class AvgOverTimeFunction extends RangeFunction {
+class CountOverTimeChunkedFunction(var count: Int = 0) extends ChunkedRangeFunction {
+  override final def reset(): Unit = { count = 0 }
+  final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
+    sampleToEmit.setValues(endTimestamp, count.toDouble)
+  }
+  final def addChunks(tsCol: Int, valueCol: Int, info: ChunkSetInfo,
+                startTime: Long, endTime: Long, queryConfig: QueryConfig): Unit = {
+    val timestampVector = info.vectorPtr(tsCol)
+    val tsReader = bv.LongBinaryVector(timestampVector)
 
-  var sum = 0d
-  var count = 0
+    // First row >= startTime, so we can just drop bit 31 (dont care if it matches exactly)
+    val startRowNum = tsReader.binarySearch(timestampVector, startTime) & 0x7fffffff
+    val endRowNum = tsReader.ceilingIndex(timestampVector, endTime)
+    val numRows = endRowNum - startRowNum + 1
```

Comment:
This short-cut will extra count the NaN values. Bummer. :(

Reply:
I'll have a special version for Double columns which filters out the last value if its a NaN.  Does that work?

Reply:
I dont think so. This is count-over-time, not last sample. The count (number of samples) in the window should exclude NaN in the count. (BTW, I am not sure we are doing this today and it may be a bug).

Reply:
I thought you said NaN was used to mark end of a time series.  When would you have NaN in the middle of a series?